Fix Main tests + small refactors

arielsvn opened the pull request:
## Issue Number

N/A

## Purpose/Implementation Notes

Our tests seem to be broken, trying to fix them here.

## Types of changes

- Bugfix (non-breaking change which fixes an issue)

## Functional tests

Local tests

## Checklist

_Put an `x` in the boxes that apply._

- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Screenshots

Please attach any screenshots that illustrate these changes.

arielsvn commented:
@kurtwheeler, this should be ready for review now and fix the `main_tests`. 

I'm gonna look into the failures of `salmon_and_api_tests` in a separate PR, it looks like the tests are not finding `tximport`.

kurtwheeler commented:
> I'm gonna look into the failures of salmon_and_api_tests in a separate PR, it looks like the tests are not finding tximport.

That'd be cool. `main_tests` are still failing though... :(

kurtwheeler commented on common/data_refinery_common/models/models.py:
```diff
@@ -719,6 +719,17 @@ def save(self, *args, **kwargs):
+        """ Marks the file as downloaded, if `filename` is not provided it will
+        be parsed from the `absolute_file_path` """
+        self.is_downloaded = True
+        self.is_archive = False
```
Woah, this seems dangerous because there's nothing about the function name or docstring that indicates it's not appropriate to call this on an archive file.

kurtwheeler commented on common/data_refinery_common/models/models.py:
```diff
@@ -738,6 +749,16 @@ def get_display_name(self):
+        return os.path.splitext(self.filename)[1].lower()
+
+    def is_blacklisted(self):
+        # TODO: extend this list.
```
I think we can just remove this TODO now. This isn't something that's on our radar or that I've seen any need for.

kurtwheeler commented on common/data_refinery_common/models/models.py:
```diff
@@ -785,7 +793,23 @@ def needs_processing(self, own_processor_id=None) -> bool:
+    def needs_processing(self, own_processor_id=None) -> bool:
+        """Returns False if original_file has been or is being processed.
+
+        Returns True otherwise.
```
:+1:

kurtwheeler commented on foreman/data_refinery_foreman/surveyor/test_end_to_end.py:
```diff
@@ -170,7 +170,7 @@ def test_array_express_redownloading(self):
 
             # Now we're going to delete one of the extracted files but not the other.
-            deleted_file = OriginalFile.objects.filter(is_archive=False)[0]
+            deleted_file = OriginalFile.objects.filter(is_archive=False).first()
```
Hmm, looks like you've traded one potential exception for another.... `first()` can return None which doesn't have `delete_local_file()`, unless you made that change for other reasons.

arielsvn replied:
added an additional assertion below this line `self.assertIsNotNone(deleted_file)`

arielsvn commented:
@kurtwheeler just addressed your comments, let me know if you see something else.

`main_tests` seem to be failing because of a new error that is also affecting other pull requests:

`urllib.error.URLError: <urlopen error ftp error: error_perm('550 Failed to change directory.',)>`

Maybe it's the same error that is affecting the salmon tests? If not, I think we should address it in a different PR, this one fixes the other errors we were getting there.
